### PR TITLE
Improve hasUpdate checks and make sure we don't offer updates to outdated versions

### PR DESCRIPTION
**Pull request by zero-24**
### Summary of Changes

Improve hasUpdate checks and make sure we don't offer downgrades to outdated versions

### Testing Instructions

- check for updates
- go to the database table #__updates 
- manually change the core update versions collum to a lower version than the current.
- go to the updater page (don't hit refresh)
- see the message to update to an lower version than installed

### Expected result

When we have an old version in the cache joomla says it runs the latest version

### Actual result

When we have an older version in the cache Joomla wants to update to that version.

### Documentation Changes Required

none

### Additional info

The behavior we build by manually changing the database table can happen in case there is some caching of the old version of the update for some reason. one of many examples (german): https://forum.joomla.de/thread/8109-joomla-update-wird-nicht-angeboten-bzw-falsches-update-3-9-10/

### CODE REVIEW

File: administrator/components/com_joomlaupdate/models/default.php [hunk 1]
```diff
@@ -164,13 +164,28 @@ public function getUpdateInformation()
...
+			// We have not found any update in the database we seem to run the latest version
 			$ret['latest'] = JVERSION;
 
 			return $ret;
 		}
 
-		$ret['latest']    = $updateObject->version;
-		$ret['hasUpdate'] = $updateObject->version != JVERSION;
+		// Check whether this is an valid update or not
+		if (version_compare($updateObject->version, JVERSION) === -1)
```
Comment (mbabker) [inline administrator/components/com_joomlaupdate/models/default.php]: Use `version_compare()` with three arguments, i.e. `version_compare($updateObject->version, JVERSION, '<=')`.  I've been writing PHP for 10 years and I couldn't tell you a thing about any of the return values of `version_compare()`, whereas specifying the comparison as the third parameter makes it crystal clear what you're intending here.

Comment (zero-24) [reply administrator/components/com_joomlaupdate/models/default.php]: has been patched thanks 👍

**Review by wilsonge — APPROVED** [review]

File: administrator/components/com_joomlaupdate/models/default.php [hunk 2]
```diff
@@ -164,13 +164,28 @@ public function getUpdateInformation()
...
 		{
+			// We have not found any update in the database we seem to run the latest version
 			$ret['latest'] = JVERSION;
 
 			return $ret;
 		}
 
-		$ret['latest']    = $updateObject->version;
-		$ret['hasUpdate'] = $updateObject->version != JVERSION;
+		// Check whether this is an valid update or not
```
Comment (richard67) [inline administrator/components/com_joomlaupdate/models/default.php]:
```suggestion
		// Check whether this is a valid update or not
```

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on a5f0ef244d7a5462fa81fd24afc82c232ace9806<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25810">issues.joomla.org/tracker/joomla-cms/25810</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on a5f0ef244d7a5462fa81fd24afc82c232ace9806<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25810">issues.joomla.org/tracker/joomla-cms/25810</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25810">issues.joomla.org/tracker/joomla-cms/25810</a>.</sub>

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on dcf8c972c69b8e5ee1e9183b9be74535dd1ff383<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25810">issues.joomla.org/tracker/joomla-cms/25810</a>.</sub>

**Comment by richard67** [comment]
Last change was only a code comment correction so @alikon 's test result is still valid.

**Comment by HLeithner** [comment]
Thank you for stop downgrading offers.

**Pull request merged**